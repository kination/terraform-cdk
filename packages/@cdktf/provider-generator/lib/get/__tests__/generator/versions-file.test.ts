import * as fs from "fs";
import * as os from "os";
import * as path from "path";
import { TerraformProviderGenerator } from "../../generator/provider-generator";
import { CodeMaker } from "codemaker";
import { ConstructsMakerTarget, Language } from "../../constructs-maker";

test("generates a versions file", async () => {
  const code = new CodeMaker();
  const workdir = fs.mkdtempSync(path.join(os.tmpdir(), "versions-file.test"));
  const spec = JSON.parse(
    fs.readFileSync(
      path.join(__dirname, "fixtures", "versions-file.test.fixture.json"),
      "utf-8"
    )
  );
  new TerraformProviderGenerator(code, spec, [
    ConstructsMakerTarget.from(
      {
        fqn: "hashicorp/aws",
        name: "aws",
        source: "hashicorp/aws",
      },
      Language.TYPESCRIPT
    ),
    ConstructsMakerTarget.from(
      {
        fqn: "hashicorp/random",
        name: "random",
        source: "hashicorp/random",
      },
      Language.TYPESCRIPT
    ),
    ConstructsMakerTarget.from(
      {
        fqn: "kreuzwerker/docker",
        name: "docker",
        source: "kreuzwerker/docker",
      },
      Language.TYPESCRIPT
    ),
  ]);
  await code.save(workdir);

  const output = fs.readFileSync(path.join(workdir, "versions.json"), "utf-8");
  expect(output).toMatchSnapshot();
});
