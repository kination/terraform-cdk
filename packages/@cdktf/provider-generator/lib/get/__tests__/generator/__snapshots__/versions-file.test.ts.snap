// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates a versions file 1`] = `
"{
  \\"registry.terraform.io/hashicorp/aws\\": \\"4.0.0\\",
  \\"registry.terraform.io/hashicorp/random\\": \\"3.1.3\\",
  \\"registry.terraform.io/kreuzwerker/docker\\": \\"2.16.0\\"
}
"
`;
